spec.html

```diff
@@ -42053,6 +42053,151 @@ <h1>Set Objects</h1>
     <p>Set objects are collections of ECMAScript language values. A distinct value may only occur once as an element of a Set's collection. Distinct values are discriminated using the SameValueZero comparison algorithm.</p>
     <p>Set objects must be implemented using either hash tables or other mechanisms that, on average, provide access times that are sublinear on the number of elements in the collection. The data structure used in this specification is only intended to describe the required observable semantics of Set objects. It is not intended to be a viable implementation model.</p>
 
+    <emu-clause id="sec-abstract-operations-for-set-objects">
+      <h1>Abstract Operations For Set Objects</h1>
+
+      <emu-clause id="sec-set-records">
+        <h1>Set Records</h1>
+        <p>An <dfn variants="Set Records">Set Record</dfn> is a Record value used to encapsulate the interface of a Set or similar object.</p>
+        <p>Set Records have the fields listed in <emu-xref href="#table-set-record-fields"></emu-xref>.</p>
+        <emu-table id="table-set-record-fields" caption="Set Record Fields">
+          <table>
+            <tr>
+              <th>
+                Field Name
+              </th>
+              <th>
+                Value
+              </th>
+              <th>
+                Meaning
+              </th>
+            </tr>
+            <tr>
+              <td>
+                [[Set]]
```

Comment:
This is going to make Ctrl+F-ing for `[[Set]]` even more annoying. Maybe `[[SetLike]]` or `[[SetObject]]` would be more appropriate anyway?